Isabelle/Chi_Calculus/Relaying.thy

```diff
@@ -13,32 +13,46 @@ lemma early_multi_receive_redundancy:
 
 lemma shortcut_redundancy:
   shows "a \<rightarrow> b \<parallel> b \<rightarrow> c \<parallel> a \<rightarrow> c \<approx>\<^sub>\<flat> a \<rightarrow> b \<parallel> b \<rightarrow> c"
-  by (fact early_multi_receive_redundancy)
+  using unidirectional_bridge_def and early_multi_receive_redundancy by simp
```

Comment:
I would prefer if we would apply defining equations (facts whose names end in `_def`) via `unfolding`. With this approach, we should be able to leave the proofs as they were before switching from `abbreviation` to `definition`, apart from the addition of the `unfolding` clause. For example, we should be able to prove `shortcut_redundancy` via `unfolding unidirectional_bridge_def by (fact early_multi_receive_redundancy`.

An advantage of this approach is that it allows us to combine the application of the defining equation with any other proof method. The approach you’re using here only works because the unfolded statement can be proved by `simp` and `simp` can also perform the unfolding. Of course, one could argue that doing the unfolding and the actual proving with a single `simp` application is more concise, but I would prefer handling the application of defining equations in a uniform way, via `unfolding`.